mne/connectivity/envelope.py

```diff
@@ -107,22 +126,28 @@ def envelope_correlation(data, combine='mean', orthogonalize="pairwise",
         corr = np.empty((n_nodes, n_nodes))
         for li, label_data in enumerate(epoch_data):
             if orthogonalize is False:  # the new code
-                label_data_orth = data_mag
-                label_data_orth_std = data_mag_std
+                label_data_orth = data_mag[li]
+                label_data_orth_std = data_mag_std[li]
             else:
                 label_data_orth = (label_data * data_conj_scaled).imag
+                np.abs(label_data_orth, out=label_data_orth)
+                if log:
+                    label_data_orth *= label_data_orth
+                    np.log(label_data_orth, out=label_data_orth)
                 label_data_orth -= np.mean(label_data_orth, axis=-1,
                                            keepdims=True)
                 label_data_orth_std = np.linalg.norm(label_data_orth, axis=-1)
                 label_data_orth_std[label_data_orth_std == 0] = 1
             # correlation is dot product divided by variances
-            corr[li] = np.dot(label_data_orth, data_mag_nomean[li])
-            corr[li] /= data_mag_std[li]
+            corr[li] = np.sum(label_data_orth * data_mag_nomean, axis=1)
+            corr[li] /= data_mag_std
             corr[li] /= label_data_orth_std
         if orthogonalize is not False:
             # Make it symmetric (it isn't at this point)
-            corr = np.abs(corr)
+            if absolute:
+                corr = np.abs(corr)
             corr = (corr.T + corr) / 2.
+            corr.flat[::corr.shape[0] + 1] = 0.
```

Comment:
Perhaps consider adding a comment for Python newcomers that this updates the diagonal efficiently.